### PR TITLE
Add query param to bypass validation in dev mode.

### LINKED ISSUE #16595: Add ability to turn off validation when in Developer Mode
## What's the issue?

It would be great to have a way to turn off the validation that happens in `src/validator-integration.js` when developer mode is turned ON. Currently this code attempts to fetch content from a URL to perform validation. When the AMP content is not actually available at the endpoint, multiple errors can be seen in the browser console.

Perhaps there could be a new viewer capability or flag that would allow bypassing of validation in developer mode.

/cc @zhangsu

### PR DESCRIPTION
**Pull request by alabiaga**
Fixes #16595

### CODE REVIEW

File: src/validator-integration.js [hunk 1]
```diff
@@ -30,8 +32,9 @@ export function maybeValidate(win) {
   if (startsWith(filename, 'about:')) { // Should only happen in tests.
     return;
   }
-
-  if (getMode().development) {
+  const bypassValidation = parseQueryString(
+      win.location.originalHash || win.location.hash)['validate'] == '0';
+  if (getMode().development && !bypassValidation) {
```
Comment (jridgewell) [inline src/validator-integration.js]: We can rearrange this to aid DCE:

```js
let load = false;
if (getMode().development) {
  load = parseQueryString()['validate'] !== '0';
}
if (load) {
  loadScript();
}

Comment (alabiaga) [reply src/validator-integration.js]: done. Thanks

File: src/validator-integration.js [hunk 2]
```diff
@@ -32,15 +32,21 @@ export function maybeValidate(win) {
   if (startsWith(filename, 'about:')) { // Should only happen in tests.
     return;
   }
-  const bypassValidation = parseQueryString(
-      win.location.originalHash || win.location.hash)['validate'] == '0';
-  if (getMode().development && !bypassValidation) {
+  let load = false;
+  if (getMode().development) {
+    load = parseQueryString(
+        win.location.originalHash || win.location.hash)['validate'] != '0';
```
Comment (choumx) [inline src/validator-integration.js]: Extract the hash to a separate variable.

```js
const hash = win.location.originalHash || win.location.hash;
validator = hash.validate !== '0';
```

File: src/validator-integration.js [hunk 3]
```diff
@@ -32,15 +32,21 @@ export function maybeValidate(win) {
   if (startsWith(filename, 'about:')) { // Should only happen in tests.
     return;
   }
-  const bypassValidation = parseQueryString(
-      win.location.originalHash || win.location.hash)['validate'] == '0';
-  if (getMode().development && !bypassValidation) {
+  let load = false;
```
Comment (choumx) [inline src/validator-integration.js]: `let validator = false;`

File: src/validator-integration.js [hunk 4]
```diff
@@ -32,15 +32,21 @@ export function maybeValidate(win) {
...
+  if (load) {
     loadScript(win.document, `${urls.cdn}/v0/validator.js`).then(() => {
       /* global amp: false */
       amp.validator.validateUrlAndLog(
           filename, win.document, getMode().filter);
     });
-  } else if (getMode().examiner) {
+  }
+
+  if (getMode().examiner) {
```
Comment (choumx) [inline src/validator-integration.js]: This should still be in an `else if` block. We don't want to possibly load both.

Comment (alabiaga) [reply src/validator-integration.js]: done.

**Pull request merged**